core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java

```diff
@@ -252,6 +254,20 @@ private static LoadingCache<AsciiString, String> buildCache(String spec) {
         return Caffeine.from(spec).build(AsciiString::toString);
     }
 
+    /**
+     * Validator for the scheme part of the URI, as defined in
+     * <a href="https://datatracker.ietf.org/doc/html/rfc3986#section-3.1">the section 3.1 of RFC3986</a>.
+     */
+    private static final Predicate<String> SCHEME_VALIDATOR =
+            scheme -> Pattern.compile("^([a-zA-Z][a-zA-Z0-9+\\-.]*)").matcher(scheme).matches();
```

Comment:
We don't need to define a predicate here. It only introduces an indirection that does nothing.

```java
private static final Pattern SCHEME_PATTERN =
        Pattern.compile("^([a-zA-Z][a-zA-Z0-9+\\-.]*)");

boolean isValidScheme(String scheme) {
    return SCHEME_PATTERN.matcher(scheme).matches();
}
```